src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java

```diff
@@ -275,107 +422,164 @@ private String getClassName(JarEntry f) {
+     * as:</p>
+     *
+     * <h4>required</h4>
+     * Failure of such a plugin will ultimately lead to the authorization
+     * framework returning failure but only after the remaining plugins have
+     * been invoked.
+     *
+     * <h4>requisite</h4>
+     * Like required, however, in the case that such a plugin returns a failure,
+     * control is directly returned to the application. The return value is that
+     * associated with the first required or requisite plugin to fail.
+     *
+     * <h4>sufficient</h4>
+     * If such a plugin succeeds and no prior required plugin has failed the
+     * authorization framework returns success to the application immediately
+     * without calling any further plugins in the stack. A failure of a
+     * sufficient plugin is ignored and processing of the plugin list continues
+     * unaffected.
+     *
+     * <p>
+     * Loaded plugins which do not occur in the configuration are appended to
+     * the list with "required" keyword. As of the nature of the class discovery
+     * this means that the order of invocation of these plugins is rather
+     * random.</p>
+     *
+     * <p>
+     * Plugins in the configuration which have not been loaded are skipped.</p>
      *
      * @param request request object
      * @param cache cache
      * @param name name
      * @param predicate predicate
      * @return true if yes
+     *
+     * @see RuntimeEnvironment#getPluginConfiguration()
      */
     @SuppressWarnings("unchecked")
-    private boolean checkAll(HttpServletRequest request, String cache, String name,
-            Predicate<IAuthorizationPlugin> predicate) {
+    private boolean checkAll(HttpServletRequest request, String cache, Nameable entity,
+            Predicate<AuthorizationPluginWrapper> predicate) {
+        Boolean val;
         Map<String, Boolean> m = (Map<String, Boolean>) request.getAttribute(cache);
 
         if (m == null) {
             m = new TreeMap<>();
+        } else if ((val = m.get(entity.getName())) != null) {
+            // cache hit
+            return val;
         }
 
-        Boolean val = m.get(name);
+        boolean overallDecision = performCheck(entity, predicate);
 
-        if (val != null) {
-            return val;
-        }
+        m.put(entity.getName(), overallDecision);
+        request.setAttribute(cache, m);
+        return overallDecision;
+    }
 
-        for (IAuthorizationPlugin plugin : getPlugins()) {
+    private boolean performCheck(Nameable entity, Predicate<AuthorizationPluginWrapper> predicate) {
+        boolean overallDecision = true;
+        for (AuthorizationPluginWrapper plugin : getPlugins()) {
+            // run the plugin's test method
             try {
-                if (!predicate.test(plugin)) {
-                    m.put(name, false);
-                    request.setAttribute(cache, m);
-                    return false;
+                boolean pluginDecision = predicate.test(plugin);
```

Comment:
I added a log output. I haven't done it before because I was little worried about the performance - on the main page this causes `plugins * (projects + groups)` number of outputs which may be a slowdown.